cli/cmd/plugin/unmanaged-cluster/cluster/minikube.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 VMware Tanzu Community Edition contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cluster
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/vmware-tanzu/community-edition/cli/cmd/plugin/unmanaged-cluster/config"
+)
+
+var minikubeBinName = "minikube"
+
+type MinikubeClusterManager struct{}
+
+// Create
+func (mkcm MinikubeClusterManager) Create(c *config.UnmanagedClusterConfig) (*KubernetesCluster, error) {
+	if !mkcm.isMinikubeRunning() {
+		// TODO: Is minikube it the current context? If not, attempt to switch to it
+
+		args := []string{
+			"start",
+		}
+
+		cmd := exec.Command(minikubeBinName, args...)
+
+		err := cmd.Run()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// readkubeconfig in bytes from configured kubeconfig.
+	// During `Prepare` this should be set to the users home kubeconfig since that's what minikube automatically uses
+	kcBytes, err := os.ReadFile(c.KubeconfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	kc := &KubernetesCluster{
+		Name:       c.ClusterName,
+		Kubeconfig: kcBytes,
+	}
+
+	return kc, nil
+}
+
+// isMinikubeRunning returns true if `minikube status` is successful
+// Running `minikube status` returns an error if the node is stopped or no profile is found
+// In those cases, we can attempt to recreate
+func (mkcm MinikubeClusterManager) isMinikubeRunning() bool {
+	args := []string{
+		"status",
+	}
+
+	cmd := exec.Command(minikubeBinName, args...)
+
+	_, err := cmd.Output()
+
+	return err == nil
+}
+
+// Get returns cluster information or returns an error
+func (mkcm MinikubeClusterManager) Get(clusterName string) (*KubernetesCluster, error) {
+	return nil, nil
+}
+
+// Delete will delete the minikube provider
+func (mkcm MinikubeClusterManager) Delete(c *config.UnmanagedClusterConfig) error {
+	args := []string{
+		"delete",
+	}
+
+	cmd := exec.Command(minikubeBinName, args...)
+
+	err := cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Prepare ensures the kubeconfig path is set to the users home kubeconfig.
+// This is the kubeconfig that minikube uses and will set it's context to when starting
+func (mkcm MinikubeClusterManager) Prepare(c *config.UnmanagedClusterConfig) error {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+
+	c.KubeconfigPath = filepath.Join(home, ".kube", "config")
```

Comment:
Rather than having this logic here, can we find a way to do resolution in the `kubeconfig` package.

We likely just need to find a way to expose this logic:

https://github.com/jpmcb/community-edition/blob/d306de9732ec423d083d3a51b59a42b61866bfc9/cli/cmd/plugin/unmanaged-cluster/kubeconfig/kubeconfig.go#L57-L63